game.js

```diff
@@ -705,6 +705,61 @@ const _gameUpdate = (timestamp, timeDiff) => {
   };
   _handlePush();
 
+  const _updateActivateAnimation = (grabUseMeshPosition) => {
+
+    let currentDistance = 100;
+    let currentAnimation = "grab_forward";
+    let distance = 0;
+
+    // Forward
+    localVector.set(0, -0.5, -0.5).applyQuaternion(localPlayer.quaternion)
+      .add(localPlayer.position);
+    currentDistance = grabUseMeshPosition.distanceTo(localVector);
+
+    // Down
+    localVector.set(0, -1.2, -0.5).applyQuaternion(localPlayer.quaternion)
+      .add(localPlayer.position);
+    distance = grabUseMeshPosition.distanceTo(localVector);
+    if (distance < currentDistance) {
+      currentDistance = distance;
+      currentAnimation = "grab_down";
+    }
+
+    // Up
+    localVector.set(0, 0.0, -0.5).applyQuaternion(localPlayer.quaternion)
+      .add(localPlayer.position);
+    distance = grabUseMeshPosition.distanceTo(localVector);
+    if (distance < currentDistance) {
+      currentDistance = distance;
+      currentAnimation = "grab_up";
+    }
+
+    // Left
+    localVector.set(-0.8, -0.5, -0.5).applyQuaternion(localPlayer.quaternion)
+      .add(localPlayer.position);
+    distance = grabUseMeshPosition.distanceTo(localVector);
+    if (distance < currentDistance) {
+      currentDistance = distance;
+      currentAnimation = "grab_left";
+    }
+    
+    // Right
+    localVector.set(0.8, -0.5, -0.5).applyQuaternion(localPlayer.quaternion)
+      .add(localPlayer.position);
+    distance = grabUseMeshPosition.distanceTo(localVector);
+    if (distance < currentDistance) {
+      currentDistance = distance;
+      currentAnimation = "grab_right";
+    }
+
+    if (localPlayer.getAction('activate')) {
+      localPlayer.getAction('activate').animationName = currentAnimation;
+    }
+
+    return {distance:currentDistance,
```

Comment:
There is no need to return here; it counts as an allocation and isn't used.

Reply:
I have removed this return statement now